### ISSUE TITLE
Bundler found conflicting requirements for the Ruby version:

### ISSUE THREAD

**Issue by Dbroqua** [gildesmarais/jekyll-loading-lazy#9]
Hello,

I'm starting with Jekyll and Ruby and I have a problem when I tries to add jekyll-loading-lazy to my project.

My Gemfile:
```ruby
source 'https://rubygems.org'

gem "bundler"
gem "jekyll"
gem "jekyll-paginate"
gem 'jekyll-tagging'
gem "jekyll-sitemap"
gem "jekyll-toc"
gem "jekyll-youtube"
gem 'jekyll-loading-lazy'
```

Output of `bundle install`:
```bash
Fetching gem metadata from https://rubygems.org/..........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler found conflicting requirements for the Ruby version:
  In Gemfile:
    Ruby

    bundler was resolved to 2.1.4, which depends on
      Ruby (>= 2.3.0)

    jekyll-loading-lazy was resolved to 0.1.1, which depends on
      Ruby (>= 2.5.8)
```

Anyone have any idea to resolve my issue? :sweat_smile: 

Regards,
DarKou

**Comment by gildesmarais** [comment]
Hi!

I spot a `gem "bundler"` in your `Gemfile`. Is that line there intentionally? Usually it's not required to require bundler inside the Gemfile.

Try removing that line and `bundle` again. Let me know if the problem remains.

Cheers,
Gil

**Comment by Dbroqua** [comment]
Hello,

Removed bundler line in my gemtile and the result:

```bash
$ bundle install 
Fetching gem metadata from https://rubygems.org/..........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler found conflicting requirements for the Ruby version:
  In Gemfile:
    Ruby

    jekyll-youtube was resolved to 1.0.0, which depends on
      jekyll was resolved to 4.1.1, which depends on
        Ruby (>= 2.4.0)

    jekyll-loading-lazy was resolved to 0.1.1, which depends on
      Ruby (>= 2.5.8)

    jekyll-toc was resolved to 0.14.0, which depends on
      Ruby (>= 2.4)

```

**Comment by Dbroqua** [comment]
For information my Ruby version 2.5.5 (Debian Buster).

Regards

**Comment by blogi-kriptovalyut** [comment]
I had the same error and just updated Ruby to 2.6

**Comment by Dbroqua** [comment]
> OK, then it's crystal-clear: the ruby version.
> 
> Ruby 2.5.8 alone had [two CVE fixes](https://www.ruby-lang.org/en/news/2020/03/31/ruby-2-5-8-released/) to its previous version. That's one reason I've set the minimum ruby version to 2.5.8. Another reason is that ruby 2.5.x is in the security maintenance phase and will EOL soon.
> 
> Debian [backported those fixes into their packaged version 2.5.5](https://metadata.ftp-master.debian.org/changelogs//main/r/ruby2.5/ruby2.5_2.5.5-3+deb10u2_changelog). They probably have their reasons to keep the version untouched.
> 
> So, what now?!
> 
> I always had serious problems with a system-wide installed ruby and thus let version managers manage the ruby version (e.g. with e.g. asdf, rbenv or rvm). Are you using a version manager already?
> 
>     1. If you do, then try install a recent ruby version with that.
> 
>     2. If you don't, well ... their might be some apt package repositories around with a recent ruby version, but I can't recommend any.

It's ok for me after installed ruby 2.7.1 with rvm.

**Comment by gildesmarais** [comment]
Hey everyone,

I'm glad to hear that using an up to date Ruby was a feasible approach.

I consider this issue as solved and thus close it.

Cheers,
Gil
